pandas/_typing.py

```diff
@@ -11,12 +11,17 @@
 from pandas.core.dtypes.generic import (
     ABCExtensionArray, ABCIndexClass, ABCSeries, ABCSparseSeries)
 
+from pandas.core.arrays import DatetimeArray, PeriodArray, TimedeltaArray
+
```

Comment:
don’t try to import these use a quoted version

Reply:
The following is the import error in the failed tests:
`ImportError: cannot import name '_INT64_DTYPE'`

The name of the module _typing.py is started with underscore. In PEP 8 ([https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles](url) ), it states

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose names start with an underscore.

Should we remove underscore from the name _typing? Or something else?

Reply:
you can’t import these names as they would cause circular imports 
and don’t need to just use ‘DatetimeArray’ and such

Reply:
`_INT64_DTYPE` is just an alias, i.e.,  `_INT64_DTYPE = np.dtype(np.int64)`. The import of numpy and the definition of `_INT64_DTYPE` are in the same file `pandas/core/dtype/common.py`.

I did a little experiment. Open a file called 'typing.py' (without underscore prefix) in the same directory as '_typing.py'. One can import `DatetimeArray` in `typing.py`.

Importing `PeriodArray` and `TImedeltaArray` also run into the same `ImportError`. Without them, one cannot define the mypy object `TypeVar`.
`DatetimeLikeArray = TypeVar('DatetimeLikeArray', DatetimeArray, PeriodArray, TimedeltaArray)`.

Reply:
@WillAyd Would you tell me your opinion? Should we change the filename of _typing.py?